tests/unit/test_charm.py

```diff
@@ -0,0 +1,145 @@
+        relation_data = {
+            "username": "banana",
+            "password": "password123",
+            "uris": "".join([uri_0, ",", uri_1]),
+        }
+        relation_id = self.create_database_relation()
+
+        self.harness.update_relation_data(
+            relation_id=relation_id, app_or_unit="mongodb-k8s", key_values=relation_data
+        )
+
+        patch_push.assert_called_with(
+            path="/etc/nrf/nrfcfg.yaml",
+            source=f'configuration:\n  DefaultPlmnId:\n    mcc: "208"\n    mnc: "93"\n  MongoDBName: free5gc\n  MongoDBUrl: { uri_0 }\n  mongoDBStreamEnable: true\n  mongodb:\n    name: free5gc\n    url: { uri_0 }\n  nfKeepAliveTime: 60\n  nfProfileExpiryEnable: true\n  sbi:\n    bindingIPv4: 0.0.0.0\n    port: 29510\n    registerIPv4: nrf\n    scheme: http\n  serviceNameList:\n  - nnrf-nfm\n  - nnrf-disc\ninfo:\n  description: NRF initial local configuration\n  version: 1.0.0\nlogger:\n  AMF:\n    ReportCaller: false\n    debugLevel: info\n  AUSF:\n    ReportCaller: false\n    debugLevel: info\n  Aper:\n    ReportCaller: false\n    debugLevel: info\n  CommonConsumerTest:\n    ReportCaller: false\n    debugLevel: info\n  FSM:\n    ReportCaller: false\n    debugLevel: info\n  MongoDBLibrary:\n    ReportCaller: false\n    debugLevel: info\n  N3IWF:\n    ReportCaller: false\n    debugLevel: info\n  NAS:\n    ReportCaller: false\n    debugLevel: info\n  NGAP:\n    ReportCaller: false\n    debugLevel: info\n  NRF:\n    ReportCaller: false\n    debugLevel: info\n  NamfComm:\n    ReportCaller: false\n    debugLevel: info\n  NamfEventExposure:\n    ReportCaller: false\n    debugLevel: info\n  NsmfPDUSession:\n    ReportCaller: false\n    debugLevel: info\n  NudrDataRepository:\n    ReportCaller: false\n    debugLevel: info\n  OpenApi:\n    ReportCaller: false\n    debugLevel: info\n  PCF:\n    ReportCaller: false\n    debugLevel: info\n  PFCP:\n    ReportCaller: false\n    debugLevel: info\n  PathUtil:\n    ReportCaller: false\n    debugLevel: info\n  SMF:\n    ReportCaller: false\n    debugLevel: info\n  UDM:\n    ReportCaller: false\n    debugLevel: info\n  UDR:\n    ReportCaller: false\n    debugLevel: info\n  WEBUI:\n    ReportCaller: false\n    debugLevel: info',  # noqa: E501
+        )
+
+    @patch("ops.model.Container.push")
+    def test_given_cant_connect_to_workload_container_when_database_created_event_then_event_is_deferred(  # noqa: E501
+        self,
+        patch_push,
+    ):
+        self.harness.set_can_connect(container="nrf", val=False)
+        relation_data = {
+            "username": "banana",
+            "password": "password123",
+            "uris": "1.2.3.4:1234,5.6.7.8:1111",
+        }
+        relation_id = self.create_database_relation()
+
+        self.harness.update_relation_data(
+            relation_id=relation_id, app_or_unit="mongodb-k8s", key_values=relation_data
+        )
+
+        patch_push.not_called()
```

Comment:
We're not testing that the event is deferred, but the push is not called. Should we change the name of the test?

Reply:
Fixed the test name